content/06.model-complexity.md

```diff
@@ -1,34 +1,34 @@
 Additional approaches like combining random forests with resampling without replacement can generate confidence intervals for the model predictions by iteratively exposing the models to incomplete datasets, mimicking real world cases where most rare disease datasets are incomplete [@doi:10.3390/genes11020226].
-Recent studies suggest that there are limitations to decision tree-based ensemble methods when applied to rare disease datasets with substantial class imbalance and label-noise [@doi:10.1007/s11634-019-00354-x; @pmid:30815073]. 
-This has led to the adoption of cascade learning, a variant of ensemble learning, where multiple methods leveraging distinct underlying assumptions are used in tandem; and augmented with algorithms like AdaBoost (boosting) to capture stable patterns existing in silver standard data [@doi:10.1109/cvpr.2001.990537; @doi:10.1007/978-3-540-75175-5_16; @doi:10.1109/icpr.2004.1334680]. 
+It should be noted here that resampling approaches are most helpful in constructing confidence intervals for algorithms that generate the same outcome everytime they are run (i.e. deterministic models).
+For non-deterministic models like decision trees that choose features at random for selecting a path to the outcome, resampling approaches can be helpful in determining robustness or stability of the model. 
+
+Recent studies suggest that there are limitations to decision tree-based ensemble methods when applied to rare disease datasets, leading to adoption of cascade learning, a variant of ensemble learning.[@doi:10.1007/s11634-019-00354-x; @pmid:30815073]
+In cascade learning, multiple methods leveraging distinct underlying assumptions are used in tandem to capture stable patterns existing in the dataset [@doi:10.1109/cvpr.2001.990537; @doi:10.1007/978-3-540-75175-5_16; @doi:10.1109/icpr.2004.1334680]. 
 For example, a cascade learning approach for identifying rare disease patients from electronic health record data utilized independent steps for feature extraction (word2vec [@arxiv:1301.3781]), preliminary prediction with ensembled decision trees, and prediction refinement using data similarity metrics [@pmid:30815073]. 
-Combining these three methods resulted in better performance than other methods when implemented on the silver standard dataset in isolation.
-Other approaches like inverse class weighting, oversampling (with methods like Synthetic Minority Oversampling Technique, abbreviated SMOTE), or cascade learning approaches that adequately represent the rarer classes are also beneficial in increasing stability of predictions while accommodating for class imbalance.
+Combining these three methods resulted in better performance than other methods when implemented on a silver standard dataset in isolation.
```

Comment:
I think this is now the first mention of "silver standard" - it might be confusing?

Reply:
```suggestion
Combining these three methods resulted in better performance than other methods in identifying rare patients in the given EHR dataset.
```
```suggestion
Combining these three methods resulted in better performance than other methods when implemented on a silver standard dataset in isolation.
```

Reply:
Here's what the introduction currently says:

> Though researchers strive to train useful and informative models, there are challenges inherent to applying ML to rare disease datasets. For example, training supervised models requires datasets where the phenotype labels have very little uncertainty (or “label-noise”) [doi:10.1093/jamia/ocw028] – termed "gold standard" datasets – but rare disease datasets often come with significant label-noise (e.g., silver standard datasets) due to limits in the current understanding of underlying biology and evolving clinical definitions of many rare diseases. Label-noise can decrease prediction accuracy and require larger samples sizes during training [doi:10.1109/tnnls.2013.2292894].

Reply:
Thanks Jackie.. I thought we had explained it before but had not looked back to confirm :(